tests/phpunit/CRM/Contribute/BAO/ContributionRecurTest.php

```diff
@@ -117,23 +117,46 @@ public function testSupportFinancialTypeChange(): void {
   }
 
   /**
-   * Test we don't change unintended fields on API edit
+   * Test we don't change unintended fields on the recurring contribution.
+   *
+   * This tests two scenarios
+   *  - editing a contribution and changing unrelated fields should leave the
+   *    currency unchanged
+   *  - Adding (or editing) contributions on the recurring should only alter
+   *    it if the contribbution is a template contribution.
    *
    * @throws \CRM_Core_Exception
    */
   public function testUpdateRecur(): void {
     $createParams = $this->_params;
     $createParams['currency'] = 'XAU';
-    $contributionRecur = $this->callAPISuccess('contribution_recur', 'create', $createParams);
+    $contributionRecur = $this->callAPISuccess('ContributionRecur', 'create', $createParams);
     $editParams = [
       'id' => $contributionRecur['id'],
       'end_date' => '+ 4 weeks',
     ];
-    $contributionRecur = $this->callAPISuccess('contribution_recur', 'create', $editParams);
-    $dao = new CRM_Contribute_BAO_ContributionRecur();
-    $dao->id = $contributionRecur['id'];
-    $dao->find(TRUE);
-    $this->assertEquals('XAU', $dao->currency, 'Edit clobbered recur currency');
+    $contributionRecurID = $this->callAPISuccess('ContributionRecur', 'create', $editParams)['id'];
+    $contributionRecur = ContributionRecur::get()->setSelect(['amount', 'currency'])->addWhere('id', '=', $contributionRecurID)->execute()->first();
+    $this->assertEquals('XAU', $contributionRecur['currency'], 'Contribution recur update should not have changed the recur currency');
+    $this->assertEquals(3, $contributionRecur['amount'], 'Contribution recur update should not have changed the recur amount');
```

Comment:
In our case we edited unrelated fields on a contribution in the series with different amounts (and actually a different currency). I put this part of the test (the create here) for future proofing as the hook only applies to edit at the moment but that could change in the future

Reply:
@artfulrobot the test ensures that if a contribution is added to a recurring contribution the amount of the recurring contribuion isn't changed (which it WOULD be if the hook were updated to fire on 'create' AND 'edit' at some point in the future - which would make it more logical albeit more problematic from my point of view - ie even more imposing assumptions

Reply:
@artfulrobot OK - I'll take a bigger look - the bigger issue here is that @mattwire doesn't agree that we shouldn't change the recurring contribution amount/ currency unless there is clear intent (ie changing a template contribution) or the calling code taking some action. 

ie if a contribution comes in for an amount that is greater for the recurring AND it comes in a way that involves an update (eg. uses the Create pending + add payment flow) then we assume in our wisdom that means that amount is the new amount for the recurring contribution in future. This feels to me a bit like going back to our old ways of making assumptions about payment processors. (There is a bit of a backstory as to why I merged that PR without doing the due dilligence I would normally do but regardless I did merge it based on the reviewer alone, which I wouldn't have normally done on a PR like that)

Reply:
@eileenmcnaughton It's not quite right that I don't agree. The scenario that you describe here would work fine *if* there was already a template contribution defined for the recurring contribution. ie. The latest contribution will *only* be used if there is no template contribution.
The unresolved issue here is what should happen when there is no template contribution - the options are 1) to create one; 2) to use the latest contribution; 3) to only update on template contributions (your suggestion).

Current behaviour is 3 but falls back to 2 if no template.

Reply:
@mattwire right - so the long standing behaviour is that when calling `repeattransaction`

1) if there is a template contribution it is used 
2) if not the most recent contribution is used as the template
3) if there is only a single line item and the recurring contribution amount differs from the contribution used as the template contribution it will override the template contribution amount.

The thing that changed is that added a new feature - a UI to edit the template contribution - such that people started to use them & then to avoid 3 causing us problems we added a hook to ensure that when it is edited the recurring contribution amount is edited to match. That makes sense because editing the template contribution is a clear intent to change the future contributions. 

However, template contributions are optional and there is no clear intent for what the impact on future payments is when a non-template contribution is edited. If someone actually wants to change the future amount the can edit the template contribution and /or recurring contribution - but at the moment we are assuming that they do - which is what changed